### PR TITLE
Mac layout for XD87

### PR DESCRIPTION
**Pull request by siano**
## Description

XD87 layout for Mac users

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd87/keymaps/mac/keymap.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_tkl_ansi(
+      KC_ESC,          KC_F1,  KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8,  KC_F9,  KC_F10, KC_F11,  KC_F12,              KC_F13,  KC_F14,  KC_F15,
+      KC_GRV,  KC_1,   KC_2,  KC_3,  KC_4,  KC_5,  KC_6,  KC_7,  KC_8,   KC_9,   KC_0,   KC_MINS, KC_EQL,  KC_BSPC,            KC_INS,  KC_HOME, KC_PGUP,
+      KC_TAB,  KC_Q,   KC_W,  KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,  KC_I,   KC_O,   KC_P,   KC_LBRC, KC_RBRC, KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,
+      KC_CAPS, KC_A,   KC_S,  KC_D,  KC_F,  KC_G,  KC_H,  KC_J,  KC_K,   KC_L,   KC_SCLN,KC_QUOT,          KC_ENT,                                       
+      KC_LSFT, KC_Z,   KC_X,  KC_C,  KC_V,  KC_B,  KC_N,  KC_M,  KC_COMM,KC_DOT, KC_SLSH,                  KC_RSFT,                     KC_UP,           
+      KC_LCTL, KC_LALT, KC_LGUI,                     KC_SPC,                       KC_APP, KC_RALT, MO(1),KC_RCTL,            KC_LEFT, KC_DOWN, KC_RGHT   ),
```
Comment (tzarc) [inline keyboards/xd87/keymaps/mac/keymap.c:32]: Should `KC_APP` be `KC_RGUI`, to match the image in the `readme.md` file?
Or perhaps the image needs updating to match this layout?

File: keyboards/xd87/keymaps/mac/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Tomasz Adamczyk <siano@sianecki.pl>
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/mac/config.h:1]: This file can be deleted.

File: keyboards/xd87/keymaps/mac/keymap.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/mac/keymap.c:23]:
```suggestion
```

File: keyboards/xd87/keymaps/mac/keymap.c [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/mac/keymap.c:74]:
```suggestion
```

File: keyboards/xd87/keymaps/mac/keymap.c (hunk 1 above)
Comment (siano) [reply keyboards/xd87/keymaps/mac/keymap.c]: You're right. Fixed now.

File: keyboards/xd87/keymaps/mac/config.h (hunk 2 above)
Comment (siano) [reply keyboards/xd87/keymaps/mac/config.h]: Done

**Pull request merged**